packages/react/src/hooks/asset/useCreateAsset.ts

```diff
@@ -12,27 +12,31 @@ import {
 import { useLivepeerProvider } from '../providers';
 
 export function useCreateAsset<TLivepeerProvider extends LivepeerProvider>(
-  options?: Partial<
-    UsePickMutationOptions<
-      Asset,
-      Error,
-      Omit<CreateAssetArgs, 'onUploadProgress'>
-    >
-  >,
+  options?: <TData = Asset[], TError = Error>(
+    provider: TLivepeerProvider,
+  ) => UsePickMutationOptions<TData, TError, CreateAssetArgs>,
 ) {
   const livepeerProvider = useLivepeerProvider<TLivepeerProvider>();
 
-  const [uploadProgress, setUploadProgress] = useState<number | undefined>(
+  const [uploadProgress, setUploadProgress] = useState<number[] | undefined>(
     undefined,
   );
 
   const internalQuery = useInternalMutation(
-    async (args: CreateAssetArgs) =>
-      createAsset<TLivepeerProvider>({
-        ...args,
-        onUploadProgress: (progress) =>
-          setUploadProgress(progress === 1 ? undefined : progress),
-      }),
+    async (args: CreateAssetArgs[]) =>
+      Promise.all(
```

Comment:
Using `Promise.all` here will imply that if one of the file uploads fails, it will stop all other file uploads (although some may have succeeded by the time another fails). Is that the behavior we want? Either way, this should be documented for users.